### PR TITLE
Wire through RequestSession for UniverseClient

### PR DESCRIPTION
**Pull request by BenWhitehead**

### CODE REVIEW

**Comment by jsancio** [comment]
LGTM. :shipit:

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 1]
```diff
@@ -202,8 +202,7 @@ object Cosmos extends FinchServer {
     // TODO(version): How are we going to get the DC/OS release version??
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: Delete this comment?

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala [hunk 2]
```diff
@@ -44,7 +42,13 @@ final class UniverseClient(implicit statsReceiver: StatsReceiver = NullStatsRece
     Option(props.getProperty("cosmos.version")).getOrElse("unknown-version")
   }
 
-  override def apply(
+  def apply(repository: PackageRepository)(implicit session: RequestSession): Future[internal.model.CosmosInternalRepository] = {
+    adminRouter.getDcosVersion().flatMap { dcosVersion =>
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala]: Do we have to look this up every time? Or can we cache it?

**Comment by cruhland** [comment]
LGTM too. Two minor comments/questions.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala (hunk 2 above)
Comment (BenWhitehead) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala]: I think we need to look it up every time. Imagine a scenario where an admin is updating the masters and a new leading master is elected, but the users request gets routed to an existing cosmos, we want to make sure that cosmos is operating relative to the clusters.

**Comment by BenWhitehead** [comment]
going to retarget master now that #374 has been merged.

**Pull request closed without merging**